### PR TITLE
#317/Add generated avatar to new chatrooms

### LINKED ISSUE #270: Redesign Avatar implementation
As they are also going to be used by chat and, it could may make sense to create a module

### LINKED ISSUE #317: Add generated avatar to new chatrooms
sic

### PR DESCRIPTION
**Pull request by gbudau**
close #270
close #317

### CODE REVIEW

File: webapp/src/shared/components/templates/AvatarPageTemplate/AvatarPageTemplate.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ type AvatarPageTemplateProps = {
   isNotFound: boolean;
   headerStatusVariant?: StatusVariant;
   title: string;
-  avatarProps: LargeAvatarProps;
+  avatarProps?: LargeAvatarProps;
```
Comment (gbudau) [inline webapp/src/shared/components/templates/AvatarPageTemplate/AvatarPageTemplate.tsx:20]: I removed the avatar from the `CreateChatroomPage` and made this prop optional.

File: webapp/src/pages/CreateChatroomPage/CreateChatroomPage.tsx [hunk 2]
```diff
@@ -77,19 +72,10 @@ export default function CreateChatroomPage() {
     register().catch((e) => console.error(e));
   };
 
-  const chatroom: Partial<Chatroom> = {};
   return (
     <AvatarPageTemplate
       isLoading={false}
       title="create chatroom"
-      avatarProps={{
```
Comment (gbudau) [inline webapp/src/pages/CreateChatroomPage/CreateChatroomPage.tsx:85]: I removed this, but we could use an editable avatar. The user uploads an image, and when they submit the form, we check if there is an avatar or generate it in the backend.

If we like the idea, we could create a ticket for this. It's nice to have, but not mandatory.

File: transcendence-app/src/chat/chat.service.ts [hunk 3]
```diff
@@ -67,20 +71,27 @@ export class ChatService {
...
+        createdAt: new Date(Date.now()),
+        avatarX: 0,
+        avatarY: 0,
+        ...chatroom,
+      },
+    );
   }
 
   async createChatroom(ownerId: string, chatroom: CreateChatroomDto) {
     const { confirmationPassword: _, ...newChat } = chatroom;
```
Comment (Damrod) [inline transcendence-app/src/chat/chat.service.ts:92]: If you don't mind, could you refer to me to some source where I can get some information on how does this _ (underscore) syntax work? Thanks

**Review by Damrod — APPROVED** [review]
LGTM, thank you!

File: transcendence-app/src/chat/chat.service.ts (hunk 3 above)
Comment (gbudau) [reply transcendence-app/src/chat/chat.service.ts:92]: If I'm not mistaken, this is a [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment), and we assign the confirmation password to the `_` variable to signal we don't use it.

**Comment by gbudau** [comment]
Thank you!

**Pull request merged**